sideboard/lib/sa/__init__.py

```diff
@@ -143,6 +143,43 @@ def process_result_value(self, value, engine):
     __all__.append('UTCDateTime')
 
 
+def check_constraint_naming_convention(constraint, table):
+    """Creates a unique name for an unnamed CheckConstraint.
+
+    The generated name is the SQL text of the CheckConstraint with
+    non-alphanumeric, non-underscore operators converted to text, and all
+    other non-alphanumeric, non-underscore substrings replaced by underscores.
+
+    If the generated name is longer than 32 characters, a uuid5 based on the
+    generated name will be returned instead.
+
+    >>> check_constraint_naming_convention(CheckConstraint('failed_logins > 3'), Table('account', MetaData()))
+    'failed_logins_gt_3'
+
+    See: http://docs.sqlalchemy.org/en/latest/core/constraints.html#configuring-constraint-naming-conventions
+    """
+    # The text of the replacements doesn't matter, so long as it's unique
+    replacements = [
+        ('||/', 'cr'), ('<=', 'le'), ('>=', 'ge'), ('<>', 'nq'), ('!=', 'ne'),
+        ('||', 'ct'), ('<<', 'ls'), ('>>', 'rs'), ('!!', 'fa'), ('|/', 'sr'),
+        ('@>', 'cn'), ('<@', 'cb'), ('&&', 'an'), ('<', 'lt'), ('=', 'eq'),
+        ('>', 'gt'), ('!', 'ex'), ('"', 'qt'), ('#', 'hs'), ('$', 'dl'),
+        ('%', 'pc'), ('&', 'am'), ('\'', 'ap'), ('(', 'lpr'), (')', 'rpr'),
+        ('*', 'as'), ('+', 'pl'), (',', 'cm'), ('-', 'da'), ('.', 'pd'),
+        ('/', 'sl'), (':', 'co'), (';', 'sc'), ('?', 'qn'), ('@', 'at'),
+        ('[', 'lbk'), ('\\', 'bs'), (']', 'rbk'), ('^', 'ca'), ('`', 'tk'),
+        ('{', 'lbc'), ('|', 'pi'), ('}', 'rbc'), ('~', 'td')]
+
+    constraint_name = str(constraint.sqltext).strip()
+    for operator, text in replacements:
+        constraint_name = constraint_name.replace(operator, text)
+
+    constraint_name = re.sub('[\W\s]+', '_', constraint_name)
+    if len(constraint_name) > 32:
+        constraint_name = uuid.uuid5(uuid.NAMESPACE_OID, str(constraint_name)).hex
```

Comment:
TIL about ``uuid5``.  I knew there were uuids other than the ``uuid4`` that I actually use, but I'd never bothered to read up on them before now.